### PR TITLE
chore(deps): migrate dirs_next to dirs

### PR DESCRIPTION
**Pull request by Legend-Master**
Reference: https://github.com/tauri-apps/tauri/issues/6063

### CODE REVIEW

**Comment by Legend-Master** [comment]
I'm not familiar with the changes dirs made in version 3 and 4 (`config_dir` is changed on Mac, `executable_dir` `$XDG_DATA_HOME` changed on Linux), need some help on those 2

**Comment by amrbashir** [comment]
> `config_dir` is changed on Mac

Looking at https://docs.rs/dirs-next/latest/dirs_next/fn.config_dir.html vs https://docs.rs/dirs/latest/dirs/fn.config_dir.html, they look the same, no?

> `executable_dir` `$XDG_DATA_HOME` changed on Linux

Same, https://docs.rs/dirs-next/latest/dirs_next/fn.executable_dir.html vs https://docs.rs/dirs/latest/dirs/fn.executable_dir.html.

Am I missing something obvious here?

**Comment by Legend-Master** [comment]
Ah, never mind the mac one, it's already changed to this directory in dirs-next 2.0, it's a breaking change in dirs 3.0, didn't look too closely at the time

About the linux one, I just don't know what does this mean, maybe there's nothing to be worried about

> BREAKING CHANGE The behavior of executable_dir has been adjusted to not depend on $XDG_DATA_HOME. Code, which assumed that setting the $XDG_DATA_HOME environment variable also impacted executable_dir if the $XDG_BIN_HOME environment variable was not set, requires adjustment.

File: .changes/dirs-next-dirs.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+"tauri": patch:breaking
+"tauri-build": patch:breaking
+"tauri-bundler": patch:breaking
+"tauri-cli": patch:breaking
+"@tauri-apps/cli": patch:breaking
```
Comment (amrbashir) [inline .changes/dirs-next-dirs.md]:
```suggestion
"tauri": patch:deps
"tauri-build": patch:deps
"tauri-bundler": patch:deps
"tauri-cli": patch:deps
"@tauri-apps/cli": patch:deps
```

**Review by amrbashir — APPROVED** [review]

**Pull request merged**